lib/ansible/modules/cloud/azure/azure_rm_lock_facts.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_lock_facts
+version_added: "2.9"
+short_description: Manage Azure locks.
+description:
+    - Create, delete an Azure lock.
+options:
+    name:
+        description:
+            - Name of the lock.
+        required: true
+    resource_id:
+        description:
+            - Id of the resource where need to manage the lock.
+            - Get this via facts module.
+            - Cannot be set mutal with C(resource_group).
+            - Query subscription if both C(resource_id) and C(resource_group) not defined.
+    resource_group:
+        description:
+            - Resource group name where need to manage the lock.
+            - The lock is in the resource group level.
+            - Cannot be set mutal with C(resource_id).
+            - Query subscription if both C(resource_id) and C(resource_group) not defined.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+
+'''
+
+EXAMPLES = '''
+- name: Get myLock details of myVM
+  azure_rm_lock_facts:
+    name: myLock
+    resource_id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/myResourceGroup/providers/Microsoft.Compute/virtualMachines/myVM
+
+- name: List locks of myVM
```

Comment:
samples on query locks on current subs and resource group. lock_facts should be used more widely than main module